data/mods/gen6/moves.ts

```diff
@@ -71,6 +71,20 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 			},
 		},
 	},
+	entrainment: {
+		inherit: true,
+		onTryHit(target, source) {
+			const bannedSourceAbilities = [
+				'flowergift', 'forecast', 'illusion', 'imposter', 'multitype', 'stancechange', 'trace', 'zenmode',
```

Comment:
So the difference here is that Zen Mode still can't be copied, but because it's not permanent, the Gen 8 code doesn't apply? I guess it would be a bit of a hack to add Zen Mode to the Gen 8 exception lists for Entrainment and Role Play so that it works automagically in Gen 6...

Reply:
hmm, I think that makes a good deal more sense to me. In exchange for minor duplication in modern gens, you don't have to inherit at all for Trace / Entrainment / Role Play, which I think is a net gain trade-off; I imagine if some new functionality were discovered with Entrainment, someone would be more likely to forget to update the mod than we lose by adding one more item in a list that could potentially be duplicated